ramips: add support for DomyWifi DM202/DM203/DW22D

DragonBluep opened the pull request:
Specifications:
* SOC: MT7620A + MT7610E
* ROM: 16 MiB spi flash (W25Q128FVSG)
* RAM: 128MiB DDR2 (W971GG6KB-25)
* WAN: 10/100M *1
* LAN: 10/100M *4
* USB: Type-A USB2.0 *1
* SD: MicroSD *1
* Button: Reset *1
* Antennas: 2.4 GHz *2 + 5 GHz *1
* TTL Baudrate: 57600
* U-Boot Recovery: IP: 10.10.10.123, Server: 10.10.10.3

Installation:
* Web UI Update
  1. Open http://192.168.10.1/upgrade.html in the browser.
  2. Rename firmware to a short name like firmware.bin and then upload it.
  3. Fill in the password column with the following content:
  password | mtd -x mIp2osnRG3qZGdIlQPh1 -r write /tmp/firmware.bin firmware
* TFTP + U-Boot
  1. Connect device with a TTL cable.
  2. Press "2" when booting to select "Load system code then write Flash via TFTP".
  3. Upload firmware by tftpd64, it will boot when write instruction is executed.

adschm commented on target/linux/ramips/dts/mt7620a_domywifi_dw22d.dtsi:
```diff
@@ -0,0 +1,151 @@
+			gpios = <&gpio1 5 GPIO_ACTIVE_HIGH>;
+		};
+
+		led_sys_green: sys_green {
```
Why not use "system" as well?

DragonBluep replied:
This name seems too long, I'll change it.

adschm replied:
?

DragonBluep replied:
If I understand you correctly, what you mean is to replace "sys_green " with "system"? The system LED has two colors so I add additional suffix.

adschm replied:
No, I mean using system_green and green:system instead of the abbreviated version. Is there any reason for the abbreviation?

DragonBluep replied:
Ok, fixed.

adschm commented on target/linux/ramips/mt7620/base-files/etc/board.d/01_leds:
```diff
@@ -81,6 +81,12 @@ dlink,dwr-960)
 	;;
+domywifi,dw22d)
+	ucidef_set_led_netdev "wlan2g" "wlan2g" "green:wlan2g" "wlan1"
+	ucidef_set_led_netdev "wlan5g" "wlan5g" "green:wlan5g" "wlan0"
```
Please use DT triggers instead.

adschm commented on target/linux/ramips/mt7620/base-files/etc/board.d/01_leds:
```diff
@@ -81,6 +81,12 @@ dlink,dwr-960)
+	ucidef_set_led_netdev "wlan2g" "wlan2g" "green:wlan2g" "wlan1"
+	ucidef_set_led_netdev "wlan5g" "wlan5g" "green:wlan5g" "wlan0"
+	ucidef_set_led_netdev "wan_link" "wan_link" "red:internet" "eth0.2" "link"
+	ucidef_set_led_netdev "wan_txrx" "wan_txrx" "green:internet" "eth0.2" "tx rx"
```
Is this vendor behavior or your invention?

DragonBluep replied:
The vendor setting is when the network is connected，show green led otherwise show red. It's not easy to achieve it so I just drop it now and user can customize it according to their preferences.

adschm replied:
Well, just use the green:internet for "link tx rx" as everyone else does in this case. And please use the proper switch-based setup for that.

adschm commented on target/linux/ramips/image/mt7620.mk:
```diff
@@ -303,6 +303,15 @@ define Device/dlink_dwr-960
+  SOC := mt7620a
+  IMAGE_SIZE := 16064k
+  DEVICE_VENDOR := DomyWifi
+  DEVICE_MODEL := DM202/DM203/DW22D
```
DEVICE_MODEL should not provide more information that the actual (image) name.

If this is for three different devices that are sold under different names, please create three "devices" for them here. (Of course, creating DTSI files etc. will help to reduce duplicate code).

DragonBluep replied:
These models are actually the same thing except the label. I just warry about these old devices will take up too much server storage resources. But if it's a unwritten rule I would like to change them.

adschm replied:
Well, one device should be one device. I never liked when the Wiki says "If you have device AAA please use the image for device XXX...."

adschm commented on target/linux/ramips/dts/mt7620a_domywifi_dw22d.dtsi:
```diff
@@ -0,0 +1,151 @@
+};
+
+&pcie0 {
+	mt76@0,0 {
```
wifi@0,0

DragonBluep replied:
After change it by your suggestions I successfully build DW22D in local host now. @adschm

adschm reviewed (changes requested):
You create a DTSI instead of a dts file here, so at least the current state of this PR is not even build-tested?

DragonBluep commented:
After change it by your suggestions I successfully build DW22D in local host now. @adschm

DragonBluep commented:
Add some device picture.

![IMG_20210416_225922](https://user-images.githubusercontent.com/70847398/115052558-a87d2900-9f10-11eb-8b4c-30276c2400dc.jpg)
![IMG_20210416_231703](https://user-images.githubusercontent.com/70847398/115052640-c054ad00-9f10-11eb-9069-79eedd543aa3.jpg)
![IMG_20210416_231859](https://user-images.githubusercontent.com/70847398/115052698-d3677d00-9f10-11eb-9e56-4cdf3349298e.jpg)

DragonBluep commented:
Add support for ethernet leds. Every port has two leds, green led can be automatically controled by ethernet pin but amber led need gpio to trigger. The vendor configuration is green led shows data and amber led shows link status.

DragonBluep commented:
Tested on master branch.

[    0.000000] Linux version 5.4.110 (dragon@debian) (gcc version 8.4.0 (OpenWrt GCC 8.4.0 r16485-cae3e37383)) #0 Fri Apr 16 12:56:22 2021
[    0.000000] Board has DDR2
[    0.000000] Analog PMU set to hw control
[    0.000000] Digital PMU set to hw control
[    0.000000] SoC Type: MediaTek MT7620A ver:2 eco:6
[    0.000000] printk: bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 00019650 (MIPS 24KEc)
[    0.000000] MIPS: machine is DomyWifi DM203
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] Primary instruction cache 64kB, VIPT, 4-way, linesize 32 bytes.
[    0.000000] Primary data cache 32kB, 4-way, PIPT, no aliases, linesize 32 bytes
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000007ffffff]
[    0.000000] On node 0 totalpages: 32768
[    0.000000]   Normal zone: 288 pages used for memmap
[    0.000000]   Normal zone: 0 pages reserved
[    0.000000]   Normal zone: 32768 pages, LIFO batch:7
[    0.000000] pcpu-alloc: s0 r0 d32768 u32768 alloc=1*32768
[    0.000000] pcpu-alloc: [0] 0 
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 32480
[    0.000000] Kernel command line: console=ttyS0,57600 rootfstype=squashfs,jffs2
[    0.000000] Dentry cache hash table entries: 16384 (order: 4, 65536 bytes, linear)
[    0.000000] Inode-cache hash table entries: 8192 (order: 3, 32768 bytes, linear)
[    0.000000] Writing ErrCtl register=0006eef1
[    0.000000] Readback ErrCtl register=0006eef1
[    0.000000] mem auto-init: stack:off, heap alloc:off, heap free:off
[    0.000000] Memory: 122348K/131072K available (4724K kernel code, 198K rwdata, 1044K rodata, 1192K init, 205K bss, 8724K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
[    0.000000] NR_IRQS: 256
[    0.000000] random: get_random_bytes called from start_kernel+0x308/0x4fc with crng_init=0
[    0.000000] CPU Clock: 580MHz
[    0.000000] clocksource: systick: mask: 0xffff max_cycles: 0xffff, max_idle_ns: 583261500 ns
[    0.000000] systick: enable autosleep mode
[    0.000000] systick: running - mult: 214748, shift: 32
[    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 6590553264 ns
[    0.000011] sched_clock: 32 bits at 290MHz, resolution 3ns, wraps every 7405115902ns
[    0.015475] Calibrating delay loop... 385.02 BogoMIPS (lpj=770048)
[    0.059566] pid_max: default: 32768 minimum: 301
[    0.068955] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.083358] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.105895] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 7645041785100000 ns
[    0.125202] futex hash table entries: 256 (order: -1, 3072 bytes, linear)
[    0.138804] pinctrl core: initialized pinctrl subsystem
[    0.149695] NET: Registered protocol family 16
[    0.421950] PCI host bridge /pcie@10140000 ranges:
[    0.431349]  MEM 0x0000000020000000..0x000000002fffffff
[    0.441719]   IO 0x0000000010160000..0x000000001016ffff
[    0.475185] rt2880_gpio 10000600.gpio: registering 24 gpios
[    0.486242] rt2880_gpio 10000600.gpio: registering 24 irq handlers
[    0.498815] rt2880_gpio 10000638.gpio: registering 16 gpios
[    0.509825] rt2880_gpio 10000638.gpio: registering 16 irq handlers
[    0.522436] rt2880_gpio 10000688.gpio: registering 1 gpios
[    0.533282] rt2880_gpio 10000688.gpio: registering 1 irq handlers
[    0.546206] PCI host bridge to bus 0000:00
[    0.554264] pci_bus 0000:00: root bus resource [mem 0x20000000-0x2fffffff]
[    0.567875] pci_bus 0000:00: root bus resource [io  0xffffffff]
[    0.579645] pci_bus 0000:00: root bus resource [??? 0x00000000 flags 0x0]
[    0.593146] pci_bus 0000:00: No busn resource found for root bus, will use [bus 00-ff]
[    0.608942] pci 0000:00:00.0: [1814:0801] type 01 class 0x060400
[    0.620873] pci 0000:00:00.0: reg 0x10: [mem 0x00000000-0x7fffffff]
[    0.633304] pci 0000:00:00.0: reg 0x14: [mem 0x20200000-0x2020ffff]
[    0.645841] pci 0000:00:00.0: supports D1
[    0.653724] pci 0000:00:00.0: PME# supported from D0 D1 D3hot
[    0.667873] pci 0000:01:00.0: [14c3:7650] type 00 class 0x028000
[    0.679757] pci 0000:01:00.0: reg 0x10: [mem 0x00000000-0x000fffff]
[    0.692293] pci 0000:01:00.0: PME# supported from D0 D3hot D3cold
[    0.704580] pci 0000:01:00.1: [14c3:8650] type 00 class 0x0d1100
[    0.716451] pci 0000:01:00.1: reg 0x10: [mem 0x00000000-0x000fffff]
[    0.728991] pci 0000:01:00.1: supports D1
[    0.736809] pci 0000:01:00.1: PME# supported from D0 D1 D3hot D3cold
[    0.751903] pci_bus 0000:01: busn_res: [bus 01-ff] end is updated to 01
[    0.764943] pci_bus 0000:00: busn_res: [bus 00-ff] end is updated to 01
[    0.778108] pci 0000:00:00.0: BAR 0: no space for [mem size 0x80000000]
[    0.791225] pci 0000:00:00.0: BAR 0: failed to assign [mem size 0x80000000]
[    0.805072] pci 0000:00:00.0: BAR 8: assigned [mem 0x20000000-0x201fffff]
[    0.818571] pci 0000:00:00.0: BAR 1: assigned [mem 0x20200000-0x2020ffff]
[    0.832081] pci 0000:01:00.0: BAR 0: assigned [mem 0x20000000-0x200fffff]
[    0.845578] pci 0000:01:00.1: BAR 0: assigned [mem 0x20100000-0x201fffff]
[    0.859075] pci 0000:00:00.0: PCI bridge to [bus 01]
[    0.868939] pci 0000:00:00.0:   bridge window [mem 0x20000000-0x201fffff]
[    0.882548] workqueue: max_active 576 requested for napi_workq is out of range, clamping between 1 and 512
[    0.907526] clocksource: Switched to clocksource systick
[    0.919454] NET: Registered protocol family 2
[    0.928919] tcp_listen_portaddr_hash hash table entries: 512 (order: 0, 4096 bytes, linear)
[    0.945489] TCP established hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.960652] TCP bind hash table entries: 1024 (order: 0, 4096 bytes, linear)
[    0.974667] TCP: Hash tables configured (established 1024 bind 1024)
[    0.987455] UDP hash table entries: 256 (order: 0, 4096 bytes, linear)
[    1.000356] UDP-Lite hash table entries: 256 (order: 0, 4096 bytes, linear)
[    1.014518] NET: Registered protocol family 1
[    1.023155] PCI: CLS 80 bytes, default 32
[    1.033991] rt-timer 10000100.timer: maximum frequency is 1220Hz
[    1.048218] workingset: timestamp_bits=14 max_order=15 bucket_order=1
[    1.069516] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    1.081028] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    1.117619] Serial: 8250/16550 driver, 2 ports, IRQ sharing disabled
[    1.131396] printk: console [ttyS0] disabled
[    1.139873] 10000c00.uartlite: ttyS0 at MMIO 0x10000c00 (irq = 20, base_baud = 2500000) is a Palmchip BK-3103
[    1.159506] printk: console [ttyS0] enabled
[    1.176090] printk: bootconsole [early0] disabled
[    1.200965] spi spi0.0: force spi mode3
[    1.209625] spi-nor spi0.0: mx25l12805d (16384 Kbytes)
[    1.219992] 4 fixed-partitions partitions found on MTD device spi0.0
[    1.232659] Creating 4 MTD partitions on "spi0.0":
[    1.242218] 0x000000000000-0x000000030000 : "bootloader"
[    1.253884] 0x000000030000-0x000000040000 : "config"
[    1.264920] 0x000000040000-0x000000050000 : "factory"
[    1.276170] 0x000000050000-0x000001000000 : "firmware"
[    1.290597] 2 uimage-fw partitions found on MTD device firmware
[    1.302475] Creating 2 MTD partitions on "firmware":
[    1.312400] 0x000000000000-0x0000001e6418 : "kernel"
[    1.323371] 0x0000001e6418-0x000000fb0000 : "rootfs"
[    1.334292] mtd: device 5 (rootfs) set to be root filesystem
[    1.347347] 1 squashfs-split partitions found on MTD device rootfs
[    1.359757] 0x000000520000-0x000000fb0000 : "rootfs_data"
[    1.372635] libphy: Fixed MDIO Bus: probed
[    1.390338] gsw: setting port4 to ephy mode
[    1.398772] mtk_soc_eth 10100000.ethernet eth0 (uninitialized): port 1 link up (100Mbps/Full duplex)
[    1.417489] mtk_soc_eth 10100000.ethernet: loaded mt7620 driver
[    1.430084] mtk_soc_eth 10100000.ethernet eth0: mediatek frame engine at 0xb0100000, irq 5
[    1.447152] rt2880_wdt 10000120.watchdog: Initialized
[    1.458750] NET: Registered protocol family 10
[    1.471788] Segment Routing with IPv6
[    1.479275] NET: Registered protocol family 17
[    1.488211] 8021q: 802.1Q VLAN Support v1.8
[    1.503284] VFS: Mounted root (squashfs filesystem) readonly on device 31:5.
[    1.524244] Freeing unused kernel memory: 1192K
[    1.533294] This architecture does not have kernel memory protection.
[    1.546134] Run /sbin/init as init process
[    1.783483] random: fast init done
[    2.541377] init: Console is alive
[    2.548591] init: - watchdog -
[    4.007936] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    4.169250] usbcore: registered new interface driver usbfs
[    4.180359] usbcore: registered new interface driver hub
[    4.191064] usbcore: registered new device driver usb
[    4.208357] ehci_hcd: USB 2.0 'Enhanced' Host Controller (EHCI) Driver
[    4.223269] ehci-fsl: Freescale EHCI Host controller driver
[    4.236271] ehci-platform: EHCI generic platform driver
[    4.257106] phy phy-usbphy.0: remote usb device wakeup disabled
[    4.268915] phy phy-usbphy.0: UTMI 16bit 30MHz
[    4.277771] ehci-platform 101c0000.ehci: EHCI Host Controller
[    4.289255] ehci-platform 101c0000.ehci: new USB bus registered, assigned bus number 1
[    4.305175] ehci-platform 101c0000.ehci: irq 26, io mem 0x101c0000
[    4.322504] ehci-platform 101c0000.ehci: USB 2.0 started, EHCI 1.00
[    4.336085] hub 1-0:1.0: USB hub found
[    4.344072] hub 1-0:1.0: 1 port detected
[    4.357584] ohci_hcd: USB 1.1 'Open' Host Controller (OHCI) Driver
[    4.371827] ohci-platform: OHCI generic platform driver
[    4.382666] ohci-platform 101c1000.ohci: Generic Platform OHCI controller
[    4.396271] ohci-platform 101c1000.ohci: new USB bus registered, assigned bus number 2
[    4.412191] ohci-platform 101c1000.ohci: irq 26, io mem 0x101c1000
[    4.449246] hub 2-0:1.0: USB hub found
[    4.457236] hub 2-0:1.0: 1 port detected
[    4.496925] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[    4.515621] init: - preinit -
[    5.966128] random: jshn: uninitialized urandom read (4 bytes read)
[    6.074317] random: jshn: uninitialized urandom read (4 bytes read)
[    6.371983] random: jshn: uninitialized urandom read (4 bytes read)
[    7.212698] 8021q: adding VLAN 0 to HW filter on device eth0
[    7.283776] urandom_read: 4 callbacks suppressed
[    7.283785] random: procd: uninitialized urandom read (4 bytes read)
[    7.461601] mtk_soc_eth 10100000.ethernet eth0: port 1 link down
[    8.383449] mtk_soc_eth 10100000.ethernet eth0: port 1 link up (100Mbps/Full duplex)
[    8.933504] mount_root: jffs2 not ready yet, using temporary tmpfs overlay
[    8.951932] urandom-seed: Seed file not found (/etc/urandom.seed)
[    9.092224] procd: - early -
[    9.098274] procd: - watchdog -
[    9.526047] procd: - watchdog -
[    9.639900] procd: - ubus -
[    9.807083] random: ubusd: uninitialized urandom read (4 bytes read)
[    9.827500] random: ubusd: uninitialized urandom read (4 bytes read)
[    9.848637] procd: - init -
[   11.127756] kmodloader: loading kernel modules from /etc/modules.d/*
[   11.191287] Loading modules backported from Linux version v5.10.16-0-gde53befa79cf
[   11.206473] Backport generated by backports.git v5.10.16-1-0-g21d2a1d2
[   11.299648] xt_time: kernel timezone is -0000
[   11.586591] PPP generic driver version 2.4.2
[   11.606666] NET: Registered protocol family 24
[   11.677028] mt76x0e 0000:01:00.0: card - bus=0x1, slot = 0x0 irq=4
[   11.689694] mt76x0e 0000:01:00.0: ASIC revision: 76100002
[   11.730987] mt76x0e 0000:01:00.0: Firmware Version: 0.1.00
[   11.768099] urngd: jent-rng init failed, err: 2
[   11.844481] mt76x0e 0000:01:00.0: EEPROM ver:01 fae:00
[   11.857623] ieee80211 phy0: Selected rate control algorithm 'minstrel_ht'
[   11.990510] rt2800_wmac 10180000.wmac: loaded eeprom from mtd device "factory"
[   12.004999] ieee80211 phy1: rt2x00_set_rt: Info - RT chipset 6352, rev 0500 detected
[   12.020477] ieee80211 phy1: rt2x00_set_rf: Info - RF chipset 7620 detected
[   12.034446] ieee80211 phy1: Selected rate control algorithm 'minstrel_ht'
[   12.043360] kmodloader: done loading kernel modules from /etc/modules.d/*
[   13.194372] mtk_soc_eth 10100000.ethernet eth0: port 1 link down
[   22.660227] random: crng init done
[   44.352840] jffs2_scan_eraseblock(): End of filesystem marker found at 0x0
[   44.396237] jffs2_build_filesystem(): unlocking the mtd device... 
[   44.396246] done.
[   44.412405] jffs2_build_filesystem(): erasing all blocks after the end marker... 
[   45.348031] 8021q: adding VLAN 0 to HW filter on device eth0
[   45.454975] br-lan: port 1(eth0.1) entered blocking state
[   45.465829] br-lan: port 1(eth0.1) entered disabled state
[   45.477018] device eth0.1 entered promiscuous mode
[   45.486618] device eth0 entered promiscuous mode
[   95.417404] done.
[   95.421304] jffs2: notice: (2013) jffs2_build_xattr_subsystem: complete building xattr subsystem, 0 of xdatum (0 unchecked, 0 orphan) and 0 of xref (0 dead, 0 orphan) found.
[   95.661336] overlayfs: upper fs does not support tmpfile.
[  101.288797] mtk_soc_eth 10100000.ethernet eth0: port 1 link up (100Mbps/Full duplex)
[  101.304497] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready
[  101.319116] br-lan: port 1(eth0.1) entered blocking state
[  101.329935] br-lan: port 1(eth0.1) entered forwarding state
[  101.341660] IPv6: ADDRCONF(NETDEV_CHANGE): eth0.2: link becomes ready
[  101.390778] IPv6: ADDRCONF(NETDEV_CHANGE): br-lan: link becomes ready

adschm commented on target/linux/ramips/mt7620/base-files/etc/board.d/01_leds:
```diff
@@ -81,6 +81,16 @@ dlink,dwr-960)
+domywifi,dm202|\
+domywifi,dm203|\
+domywifi,dw22d)
+	ucidef_set_led_switch "lan1" "lan1" "amber:lan1" "switch0" "0x02" "" "link"
```
Can you comment on what the "" is meant to achieve?

DragonBluep replied:
https://github.com/openwrt/openwrt/blob/97e820c6d61d174da1091a692546e979a5a45e42/package/base-files/files/lib/functions/uci-defaults.sh#L489
The sixth argument of function is speed mask and we don't need it so I referred to the settings of tplink,archer-d50-v1.
https://github.com/openwrt/openwrt/blob/97e820c6d61d174da1091a692546e979a5a45e42/target/linux/ath79/generic/base-files/etc/board.d/01_leds#L298
"" is just a placeholder here and it seems can work well.

adschm replied:
Ah, now I get it. This is the first time ever I see this because nobody ever tried to set mode on ucidef_set_led_switch before. (and ucidef_set_led_netdev has different options).

Please just drop the `"" "link"` arguments and set up the LEDs with default mode like all the others, as that's expected default behavior.

DragonBluep replied:
It has been dropped now.

adschm commented on target/linux/ramips/mt7620/base-files/etc/board.d/01_leds:
```diff
@@ -81,6 +81,16 @@ dlink,dwr-960)
+	ucidef_set_led_switch "lan3" "lan3" "amber:lan3" "switch0" "0x08" "" "link"
+	ucidef_set_led_switch "lan4" "lan4" "amber:lan4" "switch0" "0x10" "" "link"
+	ucidef_set_led_switch "wan" "wan" "amber:wan" "switch0" "0x01" "" "link"
+	ucidef_set_led_netdev "internet" "internet" "green:internet" "eth0.2"
```
WAN and Internet seem redundant, and the switch-based trigger is preferred. Probably it's best to simply drop the internet LED here.

DragonBluep replied:
Okay, I'll delete it.

adschm commented on target/linux/ramips/dts/mt7620a_domywifi.dtsi:
```diff
@@ -0,0 +1,183 @@
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <80000000>;
```
This will probably require m25p,fast-read;

DragonBluep replied:
okay, fixed.

adschm commented on target/linux/ramips/dts/mt7620a_domywifi.dtsi:
```diff
@@ -0,0 +1,183 @@
+	leds {
+		compatible = "gpio-leds";
+
+		system_green: system_green {
```
DT label should have led_ prefix, resulting in
`led_system_green: system_green {`
...

DragonBluep replied:
fixed, I have added it.

adschm commented on target/linux/ramips/dts/mt7620a_domywifi.dtsi:
```diff
@@ -0,0 +1,183 @@
+		led-failsafe = &system_amber;
+		led-running = &system_green;
+		led-upgrade = &system_amber;
+		label-mac-device = &wmac;
```
This does not work (label-mac-device only works for mtd-mac-address).
Either use &ethernet here if it's the same address, or use label_mac in 02_network otherwise.

DragonBluep commented:
drop useless definitions in "&ethernet", without these definitions ethernet leds can work well as usual.
```
	pinctrl-names = "default";
	pinctrl-0 = <&ephy_pins>;
```

DragonBluep commented:
Re-add pinctrl
```
	pinctrl-names = "default";
	pinctrl-0 = <&ephy_pins>;
```

adschm commented on target/linux/ramips/dts/mt7620a_domywifi.dtsi:
```diff
@@ -0,0 +1,183 @@
+	uartf_gpio {
+		groups = "uartf";
+		function = "gpio uartf";
+	};
```
Why not add this to normal gpio block?

DragonBluep replied:
Fine, fixed.

DragonBluep committed:
ramips: add support for DomyWifi DM202/DM203/DW22D

Specifications:
* SOC: MT7620A + MT7610E
* ROM: 16 MiB spi flash (W25Q128FVSG)
* RAM: 128 MiB DDR2 (W971GG6KB-25)
* WAN: 10/100M *1
* LAN: 10/100M *4
* USB: Type-A USB2.0 *1
* SD: MicroSD *1
* Button: Reset *1
* Antennas: 2.4 GHz *2 + 5 GHz *1
* TTL Baudrate: 57600
* U-Boot Recovery: IP: 10.10.10.123, Server: 10.10.10.3

Installation:
* Web UI Update
  1. Open http://192.168.10.1/upgrade.html in the browser.
  2. Rename firmware to a short name like firmware.bin and then upload it.
  3. Fill in the password column with the following content:
  password | mtd -x mIp2osnRG3qZGdIlQPh1 -r write /tmp/firmware.bin firmware
* TFTP + U-Boot
  1. Connect device with a TTL cable.
  2. Press "2" when booting to select "Load system code then write to Flash via TFTP".
  3. Upload firmware by tftpd64, it will boot when write instruction is executed.

Signed-off-by: Shiji Yang <yangshiji66@qq.com>

adschm commented:
Thanks, I will merge it later today.
